src/psyclone/tests/test_files/dynamo0p3/infrastructure/reference_element_mod.f90

```diff
@@ -0,0 +1,105 @@
+!   contributors may be used to endorse or promote products derived from
+!   this software without specific prior written permission.
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+! "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+! LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+! FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+! COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+! INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+! BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+! LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+! LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+! ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+! POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+! Modifications: A. R. Porter, STFC Daresbury Lab
+
+module reference_element_mod
+  use constants_mod, only: r_def, i_def
+  implicit none
+  private
+
+  type, abstract, public :: reference_element_type
+   contains
+
+     procedure, public :: get_number_horizontal_faces
+     procedure, public :: get_number_vertical_faces
+     procedure, public :: get_normals_to_horizontal_faces
+     procedure, public :: get_out_normals_to_horizontal_faces
+     procedure, public :: get_normals_to_vertical_faces
+     procedure, public :: get_out_normals_to_vertical_faces
+     
+  end type reference_element_type
+
+contains
+
+  integer function get_number_horizontal_faces(this)
+    implicit none
+    class(reference_element_type), intent(in)  :: this
+    get_number_horizontal_faces = 0
+  end function get_number_horizontal_faces
+
+  integer function get_number_vertical_faces(this)
+    implicit none
+    class(reference_element_type), intent(in)  :: this
+    get_number_vertical_faces = 0
+  end function get_number_vertical_faces
+  
+  subroutine get_normals_to_horizontal_faces(this, normals)
+    implicit none
+    class(reference_element_type), intent(in)  :: this
+    real(r_def), allocatable,      intent(out) :: normals(:,:)
+
+    allocate( normals(1,1) )
+  end subroutine get_normals_to_horizontal_faces
+  
+  subroutine get_out_normals_to_horizontal_faces(this, normals)
+    implicit none
+    class(reference_element_type), intent(in)  :: this
+    real(r_def), allocatable,      intent(out) :: normals(:,:)
+
+    allocate( normals(1,1) )
+  end subroutine get_out_normals_to_horizontal_faces
+  
+  subroutine get_normals_to_vertical_faces(this, normals)
+    implicit none
+    class(reference_element_type), intent(in)  :: this
+    real(r_def), allocatable,      intent(out) :: normals(:,:)
+
+    allocate( normals(1,1) )
+  end subroutine get_normals_to_vertical_faces
+  
+  subroutine get_out_normals_to_vertical_faces(this, normals)
```

Comment:
`get_outward_normals_to_vertical_faces` in LFRic implementation, please correct.